### PR TITLE
seed/seedtest, overlord/devicestate: move seed validation helper to seedtest

### PR DESCRIPTION
**Pull request by bboozzoo**
In preparation for managers level tests using the helper, move it around to the seedtest package.

### CODE REVIEW

File: seed/seedtest/seedtest.go [hunk 1]
```diff
@@ -337,3 +339,30 @@ func (s *TestingSeed20) MakeSeed(c *C, label, brandID, modelID string, modelHead
 
 	return model
 }
+
+func ValidateSeed(c *C, root, label string, trusted []asserts.Assertion) seed.Seed {
```
Comment (stolowski) [inline seed/seedtest/seedtest.go:343]: I don't know the bigger picture of the upcoming changes, but  this helper was a private helper for some tests, now it is going to be exported for reuse but it seems it has very concrete assumptions about seed, e.g. is the number of expected essential snaps always 4 with core20?

Should it be renamed if it assumes core20 seeds?

Comment (bboozzoo) [reply seed/seedtest/seedtest.go:343]: Hmm I should probably rename it to ValidateCore20Seed then

Comment (stolowski) [reply seed/seedtest/seedtest.go:343]: It seems like it's kinda general, except for assuming snapd and 4 essential snaps; maybe pass them as args to make it more reusable?

Comment (anonymouse64) [reply seed/seedtest/seedtest.go:343]: I think just renaming the helper for now is fine, we can make it more generic when it needs to be more generic I suppose

**Review by anonymouse64 — APPROVED** [review]
lgtm, but I agree that renaming the helper would be nice to make clear it assumes uc20 layout

File: seed/seedtest/seedtest.go (hunk 1 above)
Comment (bboozzoo) [reply seed/seedtest/seedtest.go]: I've tweaked the parameters to make it work with core20/core18/core16 seeds and tweaked the seedwriter tests to call the helper

**Review by stolowski — APPROVED** [review]
Thanks for the changes, looks good!

**Pull request merged**